payments/models.py

```diff
@@ -154,3 +169,309 @@ def __str__(self):
             f"{self.service} - {self.period} "
             f"({self.price_value}{'€' if self.price_unit == PriceUnits.AMOUNT else '%'})"
         )
+
+
+class OrderManager(models.Manager):
+    def berth_orders(self):
+        return self.get_queryset().filter(
+            Q(
+                Q(_product_content_type__model=BerthProduct._meta.model_name)
+                | Q(_lease_content_type__model=BerthLease._meta.model_name)
+            )
+        )
+
+    def winter_storage_orders(self):
+        return self.get_queryset().filter(
+            Q(
+                Q(_product_content_type__model=WinterStorageProduct._meta.model_name)
+                | Q(_lease_content_type__model=WinterStorageLease._meta.model_name)
+            )
+        )
+
+
+class Order(UUIDModel, TimeStampedModel):
+    customer = models.ForeignKey(
+        "customers.CustomerProfile",
+        verbose_name=_("customer"),
+        on_delete=models.CASCADE,
+        related_name="orders",
+    )
+    product = GenericForeignKey("_product_content_type", "_product_object_id")
+    lease = GenericForeignKey("_lease_content_type", "_lease_object_id")
+    status = EnumField(OrderStatus, default=OrderStatus.WAITING)
+    comment = models.TextField(blank=True, null=True)
+    base_price = models.DecimalField(
+        verbose_name=_("price"),
+        max_digits=10,
+        decimal_places=2,
+        validators=[MinValueValidator(Decimal("0.01"))],
+        blank=True,
+        null=True,
```

Comment:
not sure that the base price (be in taxul or taless) should be blankable or nullable 🤔  it is one of the main attributes of the order, and we probably want to have always defined

Reply:
I agree, but _in theory_, they will never actually be null. The reason to have them as `blank`/`null`, is so that we can receive "empty" values which then will be set based on the `product` values.

Another option would be to make them non-nullable, and call the `clean()` after the values have been set.

https://github.com/City-of-Helsinki/berth-reservations/pull/214/files#diff-409ee34417f21f35e34b41dfabf4720cR382-R385

Reply:
yeah, ok, this _base_ price can be nullable and blankable 👍